core/src/main/scala/org/broadinstitute/dsde/rawls/deltalayer/DeltaLayer.scala

```diff
@@ -116,12 +116,14 @@ class DeltaLayer(bqServiceFactory: GoogleBigQueryServiceFactory, deltaLayerWrite
   /**
     * Deletes the Delta Layer companion dataset from a given workspace.
     * @param workspace the workspace from which to delete the companion dataset
-    * @return status of dataset deletion
+    * @return true if dataset was deleted, false if it was not found (and throws on other errors)
     */
-  def deleteDataset(workspace: Workspace): IO[Boolean] = {
+  def deleteDataset(workspace: Workspace): Future[Boolean] = {
     val bqService = bqServiceFactory.getServiceForProject(workspace.googleProject)
     val datasetName = DeltaLayer.generateDatasetNameForWorkspace(workspace)
-    bqService.use(_.deleteDataset(datasetName))
+    // the GoogleBigQueryService.deleteDataset call below is already 404-safe; no need to trap
+    // for the case where the dataset doesn't exist
+    bqService.use(_.deleteDataset(datasetName)).unsafeToFuture()
```

Comment:
I moved the `.unsafeToFuture()` here instead of inside callers e.g. SnapshotService. This provides consistency between `deleteDataset` and `createDataset` - both return `Future` now.